File: src/main/java/com/innercircle/survey/domain/survey/response/SurveyResponse.java
package com.innercircle.survey.domain.survey.response;

import com.innercircle.survey.domain.AbstractEntity;
import com.innercircle.survey.domain.survey.Survey;
import jakarta.persistence.*;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Entity
@Table(name = "survey_responses")
public class SurveyResponse extends AbstractEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    private Survey survey;

    @OneToMany(mappedBy = "response")
    private List<QuestionResponse> questionResponses = new ArrayList<>();
}

File: src/main/kotlin/innerCircle/survey/entity/SurveyResponse.kt
package innerCircle.survey.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime

@Entity
data class SurveyResponse(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @ManyToOne @JoinColumn(name = "survey_version_id")
    val surveyVersion: SurveyVersion,
    val submittedAt: LocalDateTime = LocalDateTime.now()
)
